### PR TITLE
Remove the SpecRail gate workflow

### PR DESCRIPTION
**Pull request by majiayu000**
## Summary

- retire the standalone SpecRail route/PR gate, checker implementation, schemas, templates, examples, policy material, and dedicated workflow
- preserve task-event liveness and committed-whitespace checks in the required `CI Result` path
- replace inline fan-in shell logic with a fail-closed executable result checker
- harden repository tests against pytest configuration, conftest, plugin, package, and environment injection
- add mutation coverage for fan-in, scoped tests, repository tests, whitespace checks, and the single web build
- require fresh owner review for the full CI and repository-contract trust surface through CODEOWNERS and a separate review-only ruleset
- preserve Harness workflow runtime `local_review_gate` and `quality_gate_pending` states

## Why

SpecRail was a repository process layer, not part of the Harness workflow runtime. Keeping both systems created duplicate gate concepts and a large maintenance surface without affecting runtime decisions. The useful repository checks remain enforced after the removal.

The GH1454 benchmark previously ran the SpecRail pack checker, which did not validate that issue CI contract. The case is omitted until GH1768 supplies a trusted evaluator-owned historical verifier.

## Trust model

The technical `main-branch-protection` ruleset is active, requires the `CI Result` status without a bypass, and enforces strict base synchronization. `.github/CODEOWNERS` covers all GitHub configuration, hooks, checks, scripts, repository tests, and agent instruction files.

A separate active `repository-contract-review` ruleset requires one approving review, requires code-owner review on protected paths, and dismisses approvals after every push. Its PR-only bypass is scoped to the repository owner because GitHub does not allow an author to approve their own change; this PR uses that audited review bypass only after independent exact-head review and all technical checks pass. The technical CI ruleset remains independently non-bypassable.

This uses base-branch CODEOWNERS semantics and exact-head review invalidation instead of a same-name status that an untrusted GitHub Actions workflow could spoof.

## Validation

- `python3 -I -m pytest -q -c /dev/null --noconftest -p no:cacheprovider tests` — 96 passed
- Python package isolation regression includes a regular fake `pip` package and proves `-I` ignores an injected `PYTHONPATH`
- malicious `tests/__init__.py` reproduction exits candidate-side pytest with status 0 before a report; `/tests/` is therefore included in the owner-reviewed trust surface
- independent security review drove removal of the spoofable status-publisher design and validated the CODEOWNERS plus split-ruleset architecture
- `cargo test -p harness-core trust_cannot_exceed_observation_source`
- `cargo test -p harness-workflow eval_manifest`
- `cargo check --workspace --all-targets`
- `cargo clippy --workspace --all-targets -- -D warnings`
- `cargo fmt --all -- --check`
- `bash -n .githooks/pre-commit`
- `git diff --check`

A local full workspace run reached 1,355 passing server tests before a 64 MiB Docker PostgreSQL shared-memory limit caused four migration/pool failures. All four affected tests passed individually against a fresh isolated database; GitHub CI remains authoritative for the full suite.

### CODE REVIEW

File: evals/benchmarks/harness-core.toml [hunk 1]
```diff
@@ -63,7 +63,10 @@ case_id = "gh1454-scoped-ci-jobs"
 repo = "majiayu000/harness"
 issue = 1454
 base_commit = "9c0099ad458e82fd377fd20a8e288a46722762ef"
-verify_commands = ["python3 checks/check_workflow.py --repo ."]
+verify_commands = [
+  "bash -n .githooks/pre-commit",
+  "python3 -c 'from pathlib import Path as P; c=P(\".github/workflows/ci.yml\").read_text(); h=P(\".githooks/pre-commit\").read_text(); assert \"\\n  check:\\n\" not in c; assert c.count(\"bun run build\") == 1; assert all(x in c for x in (\"workspace: ${{ steps.filter.outputs.workspace }}\",\"other_crates: ${{ steps.filter.outputs.other_crates }}\",\"Compute test scope\",\"cargo test ${{ steps.scope.outputs.packages }}\",\"actions/upload-artifact@v4\",\"actions/download-artifact@v4\",\"${{ needs.web-build.result }}\")); assert all(x in h for x in (\"git diff --cached --name-only\",\"cargo clippy $scope\"))'",
+]
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply evals/benchmarks/harness-core.toml]: Addressed in db4b0b14: each benchmark assertion now reports a specific failure reason. Verified by parsing the TOML, running the exact benchmark command, pytest, rustfmt check, and git diff --check.

File: evals/benchmarks/harness-core.toml [hunk 2]
```diff
@@ -63,7 +63,10 @@ case_id = "gh1454-scoped-ci-jobs"
 repo = "majiayu000/harness"
 issue = 1454
 base_commit = "9c0099ad458e82fd377fd20a8e288a46722762ef"
-verify_commands = ["python3 checks/check_workflow.py --repo ."]
+verify_commands = [
+  "bash -n .githooks/pre-commit",
+  "python3 -c 'from pathlib import Path as P; c=P(\".github/workflows/ci.yml\").read_text(); h=P(\".githooks/pre-commit\").read_text(); assert \"\\n  check:\\n\" not in c, \"check job still exists in ci.yml\"; assert c.count(\"bun run build\") == 1, \"bun run build count mismatch\"; assert all(x in c for x in (\"workspace: ${{ steps.filter.outputs.workspace }}\",\"other_crates: ${{ steps.filter.outputs.other_crates }}\",\"Compute test scope\",\"cargo test ${{ steps.scope.outputs.packages }}\",\"actions/upload-artifact@v4\",\"actions/download-artifact@v4\",\"${{ needs.web-build.result }}\")), \"missing required ci.yml steps\"; assert all(x in h for x in (\"git diff --cached --name-only\",\"cargo clippy $scope\")), \"missing required pre-commit hooks\"'",
+]
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply evals/benchmarks/harness-core.toml]: Addressed in af778e28. The check now matches semantic expression identifiers instead of exact YAML quoting/spacing, and detects the removed check job via normalized line content. TOML parsing, the exact benchmark command, pytest, rustfmt check, and git diff --check all pass.

File: evals/benchmarks/harness-core.toml [hunk 3]
```diff
@@ -63,7 +63,10 @@ case_id = "gh1454-scoped-ci-jobs"
 repo = "majiayu000/harness"
 issue = 1454
 base_commit = "9c0099ad458e82fd377fd20a8e288a46722762ef"
-verify_commands = ["python3 checks/check_workflow.py --repo ."]
+verify_commands = [
+  "bash -n .githooks/pre-commit",
+  "python3 -c 'from pathlib import Path as P; c=P(\".github/workflows/ci.yml\").read_text(); h=P(\".githooks/pre-commit\").read_text(); assert not any(line.strip() == \"check:\" for line in c.splitlines()), \"check job still exists in ci.yml\"; assert c.count(\"bun run build\") == 1, \"bun run build count mismatch\"; assert all(x in c for x in (\"steps.filter.outputs.workspace\",\"steps.filter.outputs.other_crates\",\"Compute test scope\",\"steps.scope.outputs.packages\",\"actions/upload-artifact@v4\",\"actions/download-artifact@v4\",\"needs.web-build.result\")), \"missing required ci.yml steps\"; assert all(x in h for x in (\"git diff --cached --name-only\",\"cargo clippy $scope\")), \"missing required pre-commit hooks\"'",
+]
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply evals/benchmarks/harness-core.toml]: Addressed structurally in a7deb8ac. After three review rounds exposed the inline command as the root problem, the benchmark now invokes tests/test_ci_contract.py instead. That test reads both files with explicit UTF-8, provides focused assertion messages, checks semantic CI structure, and validates the hook with bash -n. It also runs in required Repository Checks. Focused tests 2/2 and all repository Python tests 6/6 pass.

File: tests/test_ci_contract.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        assert command in hook, f"pre-commit hook is missing: {command}"
+
+    result = subprocess.run(
+        ["bash", "-n", str(PRE_COMMIT_HOOK)],
+        cwd=ROOT,
+        check=False,
+        capture_output=True,
+        text=True,
+    )
+    assert result.returncode == 0, result.stderr
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply tests/test_ci_contract.py:723]: Fixed in a4532225. The test resolves Bash with shutil.which; when unavailable it raises SkipTest with an explicit reason instead of failing with FileNotFoundError, and the syntax check uses the resolved executable path.

File: tests/test_ci_contract.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        if line == "jobs:":
+            in_jobs = True
+            continue
+        if in_jobs and line and not line.startswith(" "):
+            break
+        if in_jobs and line.startswith("  ") and not line.startswith("    "):
+            stripped = line.strip()
+            if stripped.endswith(":") and not stripped.startswith("#"):
+                jobs.add(stripped.removesuffix(":"))
+
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply tests/test_ci_contract.py]: Fixed in a4532225. The parser now skips blank and comment-only lines before evaluating YAML indentation, and a regression test proves that a top-level comment between two jobs does not terminate the jobs mapping scan.
